integration/spark-common/src/main/scala/org/apache/spark/util/CarbonReflectionUtils.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.util
+
+import scala.reflect.runtime._
+import scala.reflect.runtime.universe._
+
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.analysis.UnresolvedRelation
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+
+import org.apache.carbondata.common.logging.LogServiceFactory
+
+/**
+ * Reflection APIs
+ */
+
+object CarbonReflectionUtils {
+
+  private val LOGGER = LogServiceFactory.getLogService(this.getClass.getCanonicalName)
+
+  private val rm = universe.runtimeMirror(getClass.getClassLoader)
+
+  /**
+   * Returns the field val from a object through reflection.
+   * @param name - name of the field being retrieved.
+   * @param obj - Object from which the field has to be retrieved.
+   * @tparam T
+   * @return
+   */
+  def getField[T: TypeTag : reflect.ClassTag](name: String, obj: T): Any = {
+    val im = rm.reflect(obj)
+
+    im.symbol.typeSignature.members.find(_.name.toString.equals(name))
+      .map(l => im.reflectField(l.asTerm).get.asInstanceOf[LogicalPlan]).getOrElse(null)
+  }
+
+  def hasField[T: TypeTag: reflect.ClassTag](name: String, obj: T): Boolean = {
```

Comment:
Instead of providing this function, you can provide a function to extract `LogicalPlan` in `InsertIntoTable` direclty, so that CarbonDecoderProcessor does not need to call `getField`, `getField` can be private

Reply:
You can add a function to return the spark version number in this file, getUnresolvedRelation can also use it.

Reply:
All places from where UnresolvedRelation is getting called, i dont have the spark session or sparkContext inorder to get the version. For e.g. in UpdateRelation of CarbonSpark2SqlParser.scala. 
Therefore used the technique of existence of field in order to judge what parameters to pass. The final motive is to be able to pass the parameters properly.

Reply:
I think we can add a parameter called "SPARK_VERSION" into CarbonProperties, when start CarbonSession, set this parameter as SparkContext.version, and then we can use this parameter in any places